streamtrace.hh

```diff
@@ -42,9 +42,12 @@ namespace cheri
 namespace disassembler
 {
 	class disassembler;
-};
+}
 namespace streamtrace
 {
+#ifndef SIZE_T_MAX
+#define SIZE_T_MAX SIZE_MAX
+#endif
```

Comment:
Where is this needed for?  We probably should be using `std::limits<size_t>::max` instead anyway...